### PR TITLE
close root window if we're in root context (fixes #84)

### LINKED ISSUE #84: Should Window.Close() close MasterWindow if in Master context?
Calling w.Close() won't close the Master window:

```
package main

import (
        "image"

        "github.com/aarzilli/nucular"
        "github.com/aarzilli/nucular/style"

        "golang.org/x/mobile/event/key"
)

func main() {
        w := nucular.NewMasterWindowSize(
                nucular.WindowScalable,
                "Test window",
                image.Point{640, 480},
                func(w *nucular.Window) {
                        w.Row(0).Dynamic(3)
                        w.Spacing(1)
                        if keys := w.Input().Keyboard.Keys; len(keys) > 0 {
                                for _, k := range keys {
                                        switch {
                                        case k.Code == key.CodeQ:
                                                w.Label("This window should be closing", "LC")
                                                w.Close()
                                        }
                                }
                        }
                },
        )
        w.SetStyle(style.FromTheme(style.WhiteTheme, 1.0))
        w.Main()
}
```

Submitting a PR that I think fixes this in a minute.

### PR DESCRIPTION
**Pull request by justledbetter**
I think this is the right way to fix this, but happy to defer to your preference!

### CODE REVIEW

**Comment by justledbetter** [comment]
In a situation where the mainwindow opens and no Popup window is created, this doesn't work unless you click on the Root window first.  Investigating further...

**Comment by justledbetter** [comment]
Okay, I think this works as intended - is there any case where Window.parent will be nil other than if we are looking at Window == MasterWindow? (see https://github.com/aarzilli/nucular/commit/d4228ee100ca016e1f9770168ece8ca7009ae8db#diff-767b2c9dfce0996eaa3a788a9155133fc51cc5b340caf6fbe63d49950b20e963R2771 line 2771)

File: nucular.go [hunk 1]
```diff
@@ -2765,6 +2765,9 @@ func (ctx *context) autoPosition() (int, int) {
 func (win *Window) Close() {
 	if win.idx != 0 {
 		win.close = true
+	} else if win.ctx.rootWindowFocus || win.parent == nil {
```
Comment (aarzilli) [inline nucular.go]: I think this should just be an else, without any condition.

Comment (justledbetter) [reply nucular.go]: Okay, let me get that fixed!

**Review by aarzilli — APPROVED** [review]

**Pull request merged**